shell/models/fleet.cattle.io.gitrepo.js

```diff
@@ -323,34 +356,16 @@ export default class GitRepo extends SteveModel {
 
   get resourcesStatuses() {
     const clusters = this.targetClusters || [];
-    const resources = this.status?.resources || [];
-    const conditions = this.status?.conditions || [];
+    const bundleDeployments = this.bundleDeployments || [];
 
     const out = [];
 
-    for (const c of clusters) {
-      const clusterBundleDeploymentResources = this.bundleDeployments
-        .find((bd) => bd.metadata?.labels?.[FLEET_ANNOTATIONS.CLUSTER] === c.metadata.name)
-        ?.status?.resources || [];
-
-      resources.forEach((r, i) => {
-        let namespacedName = r.name;
-
-        if (r.namespace) {
-          namespacedName = `${ r.namespace }:${ r.name }`;
-        }
+    for (const bd of bundleDeployments) {
+      const c = clusters.find((c) => clusterIdFromLabels(bd.metadata?.labels) === c.id);
```

Comment:
IIUC all uses of `targetClusters` do not care about ordering.

I'd recommend returning a map id->cluster from that getter instead of an array to avoid a linear search per bundle deployment.